### ISSUE TITLE
Start at login/boot on Mac? 

### ISSUE THREAD

**Issue by getthething** [janeczku/calibre-web#2967]
Hi,

Apologies if I'm posting in the wrong place/way. 

I have calibre-web up and running on my Mac running macOS 14.1.1. I'm trying to get calibre-web to run on startup so I dont have to have Terminal running all the time. I've tried setting up a .plist as a LaunchAgent but I'm running into an error thatI can't seem to find an answer to (Load failed: 5: Input/output error) 

I've also tried Platypus and created a windowless, background "app" that runs the csp.py script and that doesn't work either. No error or anything, it just does nothing when I launch it. 

I've also tried running the csp.py script using an Automator "application" and that doesn't do anything either. 

Do I have any other options before I give up? Assume I'm not a programmer, but I'm decently technical and can handle myself with clear instructions. 

Thanks!

**Comment by OzzieIsaacs** [comment]
I don‘t know macOS somI can‘t help with this

**Comment by Juhao73692** [comment]
you can create a .sh file to start calibre web and add it to login items in preference settings.
![图像](https://github.com/janeczku/calibre-web/assets/61570580/f79fc78b-5485-437d-94a1-59555ee52761)

**Comment by getthething** [comment]
I received an email that someone else had replied, however I dont see the reply here. So just in case that person (or someone else) see this, I assume the cps executable is the file I would need to create a shell script from?

**Comment by OzzieIsaacs** [comment]
Yeah, me to. From my understanding it must be python cps.py with the correct paths

**Comment by Juhao73692** [comment]
yeah. I made the comment. But somehow it disappeared.

**Comment by Juhao73692** [comment]
And if you'd like to make it windowless, you can use the module "screen". Here is the complete instruction:

## 1. Create a .sh file with commands below
`screen -dmS cps&&screen -x -S cps -p 0 -X stuff "cps\n"
`
This .sh file is expected to create a screen named "cps" and startup the calibre-web
## 2. Add executable permissions to this .sh file.
`sudo chmod +x ***.sh
`
## 3. Add the file to "Login items" in the macOS preference settings app
go to General>Login items
press "+" then select the .sh file just created.
<img width="827" alt="截屏2024-01-05 15 21 02" src="https://github.com/janeczku/calibre-web/assets/61570580/8358cce5-bbf7-481a-9735-266a88984880">

**Comment by getthething** [comment]
Thank you! This is very helpful. I think I did everything correctly, but when I try to make it executable, Terminal doesn't output anything. No errors or anything. Just goes back to the default prompt. And nothing happens to the file. I the file saved with the .sh extension. I cd'd to the directory of the file, and then ran the chmod command (replacing the ***.sh with the actual file name) This is how I have the file setup.  It is likely I don't have this part correct.

<img width="508" alt="Screenshot 2024-01-05 at 2 33 42 PM" src="https://github.com/janeczku/calibre-web/assets/61811165/510a4d62-b98f-419a-981a-778e6cdd3b5d">

**Comment by getthething** [comment]
Just in case anyone finds this and they want an answer. I fiddled with it for days. Trying to get the script to run at login, I tried setting it up as a LaunchAgent and nothing worked. I did get the script itself working and I ran fine manually. So I ended up using Platypus and created an "app" out of the script and told that to run at login. It works well, the only issue I have is that every time I login, the "app" I created asks for permission to access my documents folder. I think it has something to do with the Calibre DB though. It won't load the book covers in calibre-web until I grant permission to access the documents folder (which is where my DB is stored). Hopefully I can figure that out. 

The script that ended up working is:

<img width="570" alt="Screenshot 2024-01-09 at 11 00 07 PM" src="https://github.com/janeczku/calibre-web/assets/61811165/f8fc0ec8-05a9-4f9b-89f4-49977865da1e">
